reference/src/client/include/sa.h

```diff
@@ -44,7 +44,7 @@
  */
 #define SA_SPECIFICATION_MAJOR 3
 #define SA_SPECIFICATION_MINOR 1
-#define SA_SPECIFICATION_REVISION 0
+#define SA_SPECIFICATION_REVISION 1
```

Comment:
Incrementing the patch version should be limited to backwards compatible bug fixes. Since there is new functionality tied to some of the changes, shouldn't the minor number be incremented instead of the patch version?

Reply:
The closest thing to a new feature is this, but this doesn't change any API or functionality. It really just allows more detailed errors to be propagated up:
Issue 22: Change internal interface to return sa_status instead of bool

These are just clarifications to tests (with some bug fixes), which I wouldn't consider new functionality:
Issue 27: Process Common Encryption tests should execute in under 10 ms
Issue 24: Update tests to check all keys for SA_STATUS_OPERATION_NOT_SUPPORTED

This is a bug fix, the code incorrectly used Base64 encoding:
Issue_26: Fix SOC KeyContainer and TypeJ KeyContainer Base64URLEncoding

Reply:
Justin Shapiro
sounds more like a patch change to me.  if it's confined to tests for sure i'd go patch.  if the error messages are of value to the consumer and you'd want that highlighted in a changelog i'd go minor.

Eric Berry
My preference is to go with a patch. I don’t think there are significant enough changes to be a minor release.

Justin Shapiro
cool patch works for me

Eric Berry
I don’t the error message the customer sees, at least on the reference implementation, change at all. The changes just allow a SOC vendor, or in this case the SKB, to return more appropriate errors.
When I change code for SKB, I have to feed that back to the reference implementation.

Justin Shapiro
now you're making it sound like a feature you want to advertise.  the ability to return a more appropriate error seems like something i'd want to know about.

Eric Berry
More specifically, some of the internal interfaces used to return true or false. Most of the time true means SA_STATUS_SUCCESS and false means SA_STATUS_INTERNAL_ERROR. With the openssl implementation there were a handful of cases where false meant something else and i just returned that. When I re-implemented the internal interface to use SKB functions, the false/true wasn’t quite granular enough, I needed to be able to use the SKB failure reason to convert to a more specific secapi error so that the unit tests could pass (otherwise the unit test wouldn’t pass because it expected some specific error and got a SA_STATUS_INTERNAL_ERROR instead).
So I went through changed the key internal functions from returning bool to sa_status so that an implementor could return a more specific error if they needed to so the unit tests could pass.

Justin Shapiro
yeah i think the patch argument is stronger.  you can argue it's more of a developer note than a feature of the library.